target_chains/solana/programs/pyth-price-publisher/src/processor/initialize_publisher.rs

```diff
@@ -0,0 +1,293 @@
+use {
+    crate::{
+        accounts::{
+            buffer,
+            publisher_config,
+        },
+        ensure,
+        instruction::{
+            InitializePublisherArgs,
+            PUBLISHER_CONFIG_SEED,
+        },
+        validate::{
+            validate_authority,
+            validate_buffer,
+            validate_config,
+            validate_publisher_config,
+            validate_system,
+        },
+    },
+    solana_program::{
+        account_info::AccountInfo,
+        entrypoint::ProgramResult,
+        program::invoke_signed,
+        program_error::ProgramError,
+        pubkey::Pubkey,
+        rent::Rent,
+        system_instruction,
+        sysvar::Sysvar,
+    },
+};
+
+// Creates a publisher config account and stores the buffer account pubkey in it.
+// Verifies and initializes the buffer account.
+pub fn initialize_publisher(
```

Comment:
is it permissible for a publisher to call this instruction multiple times with different buffers? If not, what's the expected process for a publisher to resize their buffer if it is not big enough?

(treat this question as a request for documentation and put the answer in the code somewhere)

Reply:
Addressed in #1883.